src/java.desktop/macosx/native/libawt_lwawt/awt/a11y/TableAccessibility.m

```diff
@@ -0,0 +1,245 @@
+    jintArray selectedColumnNumbers = (*env)->CallObjectMethod(env, axContext, jm_getSelectedAccessibleColumns);
+    (*env)->DeleteLocalRef(env, axContext);
+    if (selectedColumnNumbers == NULL) {
+        return nil;
+    }
+    jsize arrayLen = (*env)->GetArrayLength(env, selectedColumnNumbers);
+    jint *indexsis = (*env)->GetIntArrayElements(env, selectedColumnNumbers, 0);
+    NSMutableArray<NSNumber *> *nsArraySelectedColumnNumbers = [NSMutableArray<NSNumber *> arrayWithCapacity:arrayLen];
+    for (int i = 0; i < arrayLen; i++) {
+        [nsArraySelectedColumnNumbers addObject:[NSNumber numberWithInt:indexsis[i]]];
+    }
+    (*env)->DeleteLocalRef(env, selectedColumnNumbers);
+    return [NSArray<NSNumber *> arrayWithArray:nsArraySelectedColumnNumbers];
+}
+
+- (int)accessibleRowAtIndex:(int)index
+{
+    JNIEnv *env = [ThreadUtilities getJNIEnv];
+    jobject axContext = [self axContextWithEnv:env];
+    if (axContext == NULL) return 0;
+    jclass clsInfo = (*env)->GetObjectClass(env, axContext);
+    DECLARE_METHOD_RETURN(jm_getAccessibleRowAtIndex, clsInfo, "getAccessibleRowAtIndex", "(I)I", -1);
+    jint rowAtIndex = (*env)->CallIntMethod(env, axContext, jm_getAccessibleRowAtIndex, (jint)index);
+    (*env)->DeleteLocalRef(env, axContext);
+    return (int)rowAtIndex;
+}
+
+- (int)accessibleColumnAtIndex:(int)index
+{
+    JNIEnv *env = [ThreadUtilities getJNIEnv];
+    jobject axContext = [self axContextWithEnv:env];
+    if (axContext == NULL) return 0;
+    jclass clsInfo = (*env)->GetObjectClass(env, axContext);
+    DECLARE_METHOD_RETURN(jm_getAccessibleColumnAtIndex, clsInfo, "getAccessibleColumnAtIndex", "(I)I", -1);
+    jint columnAtIndex = (*env)->CallIntMethod(env, axContext, jm_getAccessibleColumnAtIndex, (jint)index);
+    (*env)->DeleteLocalRef(env, axContext);
+    return (int)columnAtIndex;
+}
+
+- (BOOL) isAccessibleChildSelectedFromIndex:(int)index
+{
+    JNIEnv *env = [ThreadUtilities getJNIEnv];
+    jobject axContext = [self axContextWithEnv:env];
+    if (axContext == NULL) return NO;
+    jclass clsInfo = (*env)->GetObjectClass(env, axContext);
+    DECLARE_METHOD_RETURN(jm_isAccessibleChildSelected, clsInfo, "isAccessibleChildSelected", "(I)Z", NO);
+    jboolean isAccessibleChildSelected = (*env)->CallIntMethod(env, axContext, jm_isAccessibleChildSelected, (jint)index);
+    (*env)->DeleteLocalRef(env, axContext);
+    return isAccessibleChildSelected;
+}
+
+// NSAccessibilityElement protocol methods
+
+- (NSArray *)accessibilityChildren
+{
+    NSArray *children = [self accessibilityRows];
+    NSArray *columns = [self accessibilityColumns];
+    NSMutableArray *results = [NSMutableArray arrayWithCapacity:[children count] + [columns count]];
+    [results addObjectsFromArray:children];
+    [results addObjectsFromArray:columns];
+    return [NSArray arrayWithArray:results];
+}
+
+- (NSArray *)accessibilitySelectedChildren
+{
+    return [self accessibilitySelectedRows];
```

Comment:
Does it mean that the children of the table are rows and columns and not "cells".? How it will work if only one cell is selected?

Reply:
As @ azuev-java canceled below this is not functional regression.
However, cell navigation will be implemented later in [JDK-8271071](https://bugs.openjdk.java.net/browse/JDK-8271071).

Reply:
ok

Reply:
Ok